### PR TITLE
extmod,unix: Show compiler info in REPL banner.

### PR DESCRIPTION
**Pull request by stinos**
When testing out things I normally work in a single codebase, shared across different compiler/platform combinations (e.g. unix+clang+gcc, windows+msvc, migw+clang+gcc) which makes it easy to lose track of which micropython executable is being started. This fixes that just like CPython does.

It's not strictly needed to expose mp_platform_info_python_compiler_obj via a header, a less intrusive alternative would be for `do_repl` to get the `platform` module directly, but I'm not sure if that pattern is used somewhere else? Would look like:
```
extern const mp_obj_module_t mp_module_uplatform;
mp_obj_t fun = mp_obj_dict_get(mp_module_uplatform.globals, MP_ROM_QSTR(MP_QSTR_python_compiler));
mp_obj_t info_str = mp_call_function_0(fun);
...
```

Result looks like:
> MicroPython v1.18 on 2022-02-17; win32 [MSC v.1930 64 bit] version

or

> MicroPython v1.18 on 2022-02-17; win32 [GCC 11.2.0] version

### CODE REVIEW

**Comment by dpgeorge** [comment]
This approach looks fine to me, it's OK to expose that uplatform object.

**Comment by dpgeorge** [comment]
Maybe it makes more sense to move the definition of `PLATFORM_COMPILER` to `moduplatform.h` so it can be used directly?  That would be more efficient and use less code (and still be able to reuse the same string because the compiler is smart enough for that).

**Comment by stinos** [comment]
Considered that yes, but shouldn't we then just move the other macros from .c to .h already as well (`PLATFORM_ARCH` etc)?

**Comment by dpgeorge** [comment]
I don't see why not, that would be useful.  They'd need to be prefixed with something unique though, like `MICROPY_PLATFORM_xxx`?

File: extmod/moduplatform.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2022 Damien P. George
```
Comment (dpgeorge) [inline extmod/moduplatform.h:6]: Would be better to copy the copyright line from `moduplatform.c`, because the macros below were copied out of that file.

**Review by dpgeorge — CHANGES_REQUESTED** [review]

File: extmod/moduplatform.h (hunk 1 above)
Comment (stinos) [reply extmod/moduplatform.h]: Ok.

**Comment by dpgeorge** [comment]
Thanks for updating, merged in 49934fcf8b0b2bd906a96e1ff57b724d8dd6d93e and 5f50f4a130e60c4bc5f4fbf58ac849dcaf764e72

**Pull request closed without merging**